### PR TITLE
Do not let PIL manage the file handle

### PR DESCRIPTION
**Pull request by blink1073**
Inspired by https://github.com/matplotlib/matplotlib/pull/4619.

This should clear up the `unclosed file` warnings we've been getting.

### CODE REVIEW

**Comment by JDWarner** [comment]
I love the context manager approach here. Will wait for at least one of the CI builds.

**Comment by blink1073** [comment]
This one will have to wait for travis, since we do not fail on warnings with Appveyor.

**Comment by blink1073** [comment]
@JDWarner, this one is ready to go.

File: skimage/io/_plugins/pil_plugin.py [hunk 1]
```diff
@@ -43,18 +43,12 @@ def imread(fname, dtype=None, img_num=None, **kwargs):
...
-        error_message = ('Could not load "%s" \n'
-            'Reason: "%s"\n'
-            'Please see documentation at: %s') % (fname, pillow_error_message, site)
-        raise ValueError(error_message)
+    if isinstance(fname, string_types):
+        with open(fname, 'rb') as f:
+            im = Image.open(f)
+            return pil_to_ndarray(im, dtype=dtype, img_num=img_num)
     else:
+        im = Image.open(fname)
```
Comment (JDWarner) [inline skimage/io/_plugins/pil_plugin.py]: Couldn't this branch still go unclosed? I think this needs a context manager to match the prior `if` block.

Comment (blink1073) [inline skimage/io/_plugins/pil_plugin.py]: This is for a file-like object that is already open.

**Comment by JDWarner** [comment]
@blink1073 One minor point, then happy to merge!

File: skimage/io/_plugins/pil_plugin.py (hunk 1 above)
Comment (JDWarner) [inline skimage/io/_plugins/pil_plugin.py]: I see that now - thanks.

**Pull request merged**